code/modules/client/client procs.dm

```diff
@@ -362,14 +362,17 @@
 
 /client/proc/log_client_to_db()
 
-	if( IsGuestKey(src.key) )
+	if( IsGuestKey(key) )
 		return
 
+	// Isn't the ckey sanitized by default?
+	var/sql_ckey = sanitizeSQL(ckey)
```

Comment:
Are there details about how BYOND's auth system works somewhere? It seems *unlikely* that the ckey would be provided by the auth system rather than generated locally, considering `ckey` is just `ckey(key)`, but it *is* BYOND we're talking about here.

If it's really worth worrying about, it would probably be easier to just add a `ckey != ckey(key)` check that boots clients with malformed ckeys.

Also, just noticed, I was wrong about the underscores and dashes, since `ckeyEx` makes a special exception for them. So, just lowercase alphanumeric characters, then.